nuxeo-services/login/nuxeo-platform-login-keycloak/src/main/java/org/nuxeo/ecm/platform/ui/web/keycloak/KeycloakAuthenticationPlugin.java

```diff
@@ -173,10 +173,12 @@ private KeycloakUserInfo getKeycloakUserInfo(AccessToken token) {
      */
     private Set<String> getRoles(AccessToken token, String keycloakNuxeoApp) {
         Set<String> allRoles = new HashSet<>();
-        Set<String> roles = token.getRealmAccess().getRoles();
-        if (roles != null) {
-            allRoles.addAll(roles);
+
+        AccessToken.Access realmAccess = token.getRealmAccess();
+        if (realmAccess != null && realmAccess.getRoles() != null) {
+            allRoles.addAll(realmAccess.getRoles());
```

Comment:
Another possible variant, to avoid calling `realmAccess.getRoles()` twice in case it's costly (I don't know), is:

        Set<String> roles;
        if (realmAccess != null && (roles = realmAccess.getRoles()) != null) {
              allRoles.addAll(roles);
        }

Up to you if you want to change it.